src/Compilers/VisualBasic/Portable/Binding/Binder_Attributes.vb

```diff
@@ -794,7 +794,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                                    Not _binder.IsValidTypeForAttributeArgument(conv.Operand.Type) Then
 
                                     If Not conv.HasErrors Then
-                                        ReportDiagnostic(diagBag, conv.Operand.Syntax, ERRID.ERR_RequiredAttributeConstConversion2, conv.Operand.Type, conv.Type)
+                                        ' BC30934: Conversion from '{0}' to '{1}' cannot occur in a constant expression used as an argument to an attribute.
+                                        Dim firstArgument As Object = If(conv.Operand.Type Is Nothing, DirectCast("Nothing", Object), DirectCast(conv.Operand.Type, Object))
+                                        ReportDiagnostic(diagBag, conv.Operand.Syntax, ERRID.ERR_RequiredAttributeConstConversion2, firstArgument, conv.Type)
```

Comment:
>firstArgument [](http://example.com/codeflow?start=132&length=13)

For the test scenario, the legacy native compiler reports:
```
error BC30934: Conversion from 'Object' to 'Integer?' cannot occur in a constant expression used as an argument to an attribute.
<My(Nothing)>
    ~~~~~~~
 ``` 

Even though, at the end this error is not reported for the test scenario by the current compiler, simply doing substitution like
```If(conv.Operand.Type, _binder.Compilation.GetSpecialType(SpecialType.System_Object))``` should align the behavior.